### PR TITLE
An aktualizr-lite callback container example

### PR DESCRIPTION
**Pull request by Tim-Anderson**
This is an example container showing how to get callbacks into a container.

Signed-off-by: Tim Anderson <tim.anderson@foundries.io>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,4 +11,8 @@ Runs the chromium browser inside a container and displays a website.
 ## k3s
 Runs a minimal kubernetes installation using Rancher's k3s project. 
 
+## aklite-callback-example
+This is a minimal container that demonstrates how to get aktualizr-lite callbacks
+into a container in an asyncrhronous way.
```
Comment (doanac) [inline README.md:16]: typo - `asyncrhronous`

File: aklite-callback-example/handle-callback-example.sh [hunk 2]
```diff
@@ -0,0 +1,105 @@
+#!/bin/sh
+################################################################
+#
+# This is a simple docker used to show hanling of
```
Comment (doanac) [inline aklite-callback-example/handle-callback-example.sh:4]: "hanling"

File: aklite-callback-example/handle-callback-example.sh [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+#
+# Setup function for activating the callback mechanism
+# with aktualizr-lite
+#
+function setup()
+{
+        #
+        # Make sure callback file has not changed, otherwise change it
+        #
+	old="$(md5sum ${CBFILE} 2> /dev/null | cut -d ' ' -f 1)"
```
Comment (doanac) [inline aklite-callback-example/handle-callback-example.sh:39]: tabs/spaces?

File: aklite-callback-example/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,6 @@
+FROM alpine:3.12
+
+COPY example-callback.sh /
+COPY handle-callback-example.sh /
+
+ENTRYPOINT ["/handle-callback-example.sh"]
```
Comment (doanac) [inline aklite-callback-example/Dockerfile:6]: I suppose this is probably how we should show for an example. However, you could just bind mount these files in the docker-compose.yml and not have to bother with building a custom container - just use alpine:3.12

**Review by doanac — APPROVED** [review]
all the high-level "how it fits in a factory" stuff looks fine.

the code looks fine, but I'll defer final judgement to a platform person :)

File: aklite-callback-example/handle-callback-example.sh [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+                # - download-post         return: OK or FAILED
+                # - install-pre           return: none
+                # - install-post          return: NEEDS_COMPLETION, OK, or FAILED
+
+        case $msg in
+        check-for-update-pre)
+            log "check-for-update-pre"
+            ;;
+        check-for-update-post)
+            log "check-for-update-pre"
```
Comment (mwasilew) [inline aklite-callback-example/handle-callback-example.sh:81]: should be 'check-for-update-post'

**Pull request merged**